File: rb/features.rb
# IpGeolocationApi3 SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module IpGeolocationApi3Features
  def self.make_feature(name)
    case name
    when "base"
      IpGeolocationApi3BaseFeature.new
    when "test"
      IpGeolocationApi3TestFeature.new
    else
      IpGeolocationApi3BaseFeature.new
    end
  end
end

File: py/features.py
# IpGeolocationApi3 SDK feature factory

from feature.base_feature import IpGeolocationApi3BaseFeature
from feature.test_feature import IpGeolocationApi3TestFeature


def _make_feature(name):
    features = {
        "base": lambda: IpGeolocationApi3BaseFeature(),
        "test": lambda: IpGeolocationApi3TestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
